File: src/main/java/edu/marmara/repository/impl/StudentRepositoryImpl.java
package edu.marmara.repository.impl;

import edu.marmara.model.School;
import edu.marmara.model.Student;
import edu.marmara.repository.StudentRepository;

import java.util.ArrayList;
import java.util.Objects;

public class StudentRepositoryImpl implements StudentRepository {

    // todo: Update the corresponding json file
    // todo: Save the schedule, parse the saved schedule on re-run
    @Override
    public void save(Student student) {
        School school = School.getInstance();

        if (school.getStudents() == null) {
            school.setStudents(new ArrayList<>());
        }

        school.addStudent(student);
    }

    @Override
    public Student findByStudentId(Long studentId) {
        School school = School.getInstance();

        if (school.getStudents() == null) {
            return null;
        }


        for (Student student : school.getStudents()) {
            if (Objects.equals(student.getStudentId(), studentId)) {
                return student;
            }
        }

        return null;
    }
}

File: python/src/main/marmara/repository/impl/StudentRepositoryImpl.py
from typing import List

from src.main.marmara.model import School, Student
from src.main.marmara.repository import StudentRepository


class StudentRepositoryImpl(StudentRepository):
    # todo: Update the corresponding json file
    # todo: Save the schedule, parse the saved schedule on re-run
    def save(self, student: Student) -> None:
        school = School.get_instance()

        if school.students is None:
            school.students = []

        school.add_student(student)

    def find_by_student_id(self, student_id: int) -> Student:
        school = School.get_instance()

        if school.students is None:
            return None

        for student in school.students:
            if student.student_id == student_id:
                return student

        return None
